test/runner/job_test.exs

```diff
@@ -45,4 +19,240 @@ defmodule ElixirBench.Runner.JobTest do
     assert job.context.worker_num_cores in 1..64
     assert job.context.worker_os in [:Linux, :Windows, :macOS]
   end
+
+  describe "start_job/2" do
+    test "run job and return results given execute function" do
+      task_fun = fn job ->
+        {:ok, %{job | log: "success", status: 0}}
+      end
+
+      {:ok, job} = Job.start_job(job_fixture(), task_fun)
+
+      assert %{log: "success", status: 0} = job
+    end
+
+    test "not start job if there is other job running" do
```

Comment:
ah I see you already added the test here you just need to make the function private again I think

Reply:
Yup